tools/topology/topology2/include/components/src_format.conf

```diff
@@ -1,369 +1,240 @@
-				out_valid_bit_depth     24
-				obs "$[($out_channels * (($[($out_rate + 999)] / 1000) + 4)) * ($out_bit_depth / 8)]"
-				}
-
-			Object.Base.audio_format.33 {
-				in_rate                 88200
-				in_bit_depth            32
-				in_valid_bit_depth      32
-				out_bit_depth           32
-				out_valid_bit_depth     32
-				obs "$[($out_channels * (($[($out_rate + 999)] / 1000) + 4)) * ($out_bit_depth / 8)]"
-				}
-
-			# 96khz input
-			Object.Base.audio_format.34 {
-				in_rate                 96000
-				in_bit_depth            16
-				in_valid_bit_depth      16
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				}
-
-			Object.Base.audio_format.35 {
-				in_rate                 96000
-				in_bit_depth            32
-				in_valid_bit_depth      24
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				}
-
-			Object.Base.audio_format.36 {
-				in_rate                 96000
-				in_bit_depth            32
-				in_valid_bit_depth      32
-				out_bit_depth           32
-				out_valid_bit_depth     32
-				}
-
-			# 176.4khz input
-			Object.Base.audio_format.37 {
-				in_rate                 176400
-				in_bit_depth            16
-				in_valid_bit_depth      16
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				obs "$[($out_channels * (($[($out_rate + 999)] / 1000) + 4)) * ($out_bit_depth / 8)]"
-				}
-
-			Object.Base.audio_format.38 {
-				in_rate                 176400
-				in_bit_depth            32
-				in_valid_bit_depth      24
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				obs "$[($out_channels * (($[($out_rate + 999)] / 1000) + 4)) * ($out_bit_depth / 8)]"
-				}
-
-			Object.Base.audio_format.39 {
-				in_rate                 176400
-				in_bit_depth            32
-				in_valid_bit_depth      32
-				out_bit_depth           32
-				out_valid_bit_depth     32
-				obs "$[($out_channels * (($[($out_rate + 999)] / 1000) + 4)) * ($out_bit_depth / 8)]"
-				}
-
-			# 192khz input
-			Object.Base.audio_format.40 {
-				in_rate                 192000
-				in_bit_depth            16
-				in_valid_bit_depth      16
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				}
-
-			Object.Base.audio_format.41 {
-				in_rate                 192000
-				in_bit_depth            32
-				in_valid_bit_depth      24
-				out_bit_depth           32
-				out_valid_bit_depth     24
-				}
-
-			Object.Base.audio_format.42 {
-				in_rate                 192000
-				in_bit_depth            32
-				in_valid_bit_depth      32
-				out_bit_depth           32
-				out_valid_bit_depth     32
-				}
+			Object.Base.input_audio_format [
+				# 8khz input
+				{
+					in_rate                 8000
+					in_bit_depth            16
+					in_valid_bit_depth      16
+				}
+				{
+					in_rate                 8000
+					in_bit_depth            32
+					in_valid_bit_depth      24
+				}
+				{
+					in_bit_depth            32
+					in_valid_bit_depth      32
+				}
```

Comment:
is there any value in having all these 3 formats? In what case would the first and second be used? Usually there's a copier that will convert to S32_LE, no?

Reply:
Yes, we could narrow the scope of formats. I was also thinking to rename this to have "playback" in .conf file name. With formats reduced maybe src_fomat_host_mixin_playback.conf, or something. I will need other version for capture. What do you think?